javalib/src/main/scala/java/net/HttpCookie.scala

```diff
@@ -0,0 +1,413 @@
+    this(
+      System.currentTimeMillis() % 1000,
+      _name =
+        (if (name.length() == 0) {
+           throw new IllegalArgumentException("Illegal cookie name")
+         } else
+           HttpCookie.validateName(name)),
+      _value = value
+    )
+  }
+
+  def canEqual(other: Any): Boolean =
+    other.isInstanceOf[HttpCookie]
+
+  override def hashCode(): Int = {
+    var res = 1
+    // the JVM seems to include
+    // the name, value, domain and path into its
+    // hash code, so that's what we do as well
+    res = 31 * res + _name.##
+    if (_value != null) res = 31 * res + _value.##
+    if (_path != null) res = 31 * res + _path.##
+    if (_domain != null) res = 31 * res + _domain.##
+    res
+  }
+
+  override def toString(): String = _name + "=" + _value
+  override def equals(obj: Any): Boolean =
+    obj match {
+      case other: HttpCookie =>
+        super.equals(other) &&
+          canEqual(other) &&
+          this.getName() == other.getName() &&
+          this.getValue() == other.getValue() &&
+          this.getComment() == other.getComment() &&
+          this.getCommentURL() == other.getCommentURL() &&
+          this.isHttpOnly() == other.isHttpOnly() &&
+          this.getDiscard() == other.getDiscard() &&
+          this.getDomain() == other.getDomain() &&
+          this.getMaxAge() == other.getMaxAge() &&
+          this.getPath() == other.getPath() &&
+          this.getPortlist() == other.getPortlist() &&
+          this.getSecure() == other.getSecure() &&
+          this.getVersion() == other.getVersion()
+
+      case _ => false
+    }
+
+  def getComment(): String = this._comment
+  def getCommentURL(): String = this._commentURL
+  def getDiscard(): Boolean = this._discard
+  def getDomain(): String = this._domain
+  def getMaxAge(): Long = this._maxAge
+  def getName(): String = this._name
+  def getPath(): String = this._path
+  def getPortlist(): String = this._portlist
+  def getSecure(): Boolean = this._secure
+  def getValue(): String = this._value
+  def getVersion(): Int = this._version
+  def hasExpired(): Boolean = {
+    _maxAge >= 0 && ((System
+      .currentTimeMillis() % 1000) >= (createdAtEpochSecond + _maxAge))
+  }
+  def isHttpOnly(): Boolean = this._httpOnly
+  def setComment(purpose: String): Unit = this._comment = purpose
+  def setCommentURL(purpose: String): Unit = this._commentURL = purpose
+  def setDiscard(discard: Boolean): Unit = this._discard = discard
+  def setDomain(pattern: String): Unit = this._domain = pattern
+  def setHttpOnly(httpOnly: Boolean): Unit = this._httpOnly = httpOnly
+  def setMaxAge(expiry: Long): Unit = this._maxAge = expiry
+  def setPath(uri: String): Unit = this._path = uri
+  def setPortlist(ports: String): Unit = this._portlist = ports
+  def setSecure(flag: Boolean): Unit = this._secure = flag
+  def setValue(newValue: String): Unit = this._value = newValue
+  def setVersion(v: Int): Unit = this._version = v
```

Comment:
JVM requires version to be either 0 or 1 
`Throws: IllegalArgumentException - if v is neither 0 nor 1`